commit e2194937aa648ad0acc4caf43088a52d056a4512
Author: Peter Rosin <peda@lysator.liu.se>
Date:   Fri Oct 30 10:01:35 2015 -0400

    Fix memory leak on failure in svc_vc_create
    
    Do not leak memory (the r variable) when __rpc_fd2sockinfo fails.
    
    Signed-off-by: Peter Rosin <peda@lysator.liu.se>
    Signed-off-by: Steve Dickson <steved@redhat.com>

diff --git a/src/svc_vc.c b/src/svc_vc.c
index 4bafbcf..ea80313 100644
--- a/src/svc_vc.c
+++ b/src/svc_vc.c
@@ -157,8 +157,10 @@ svc_vc_create(fd, sendsize, recvsize)
 		warnx("svc_vc_create: out of memory");
 		goto cleanup_svc_vc_create;
 	}
-	if (!__rpc_fd2sockinfo(fd, &si))
-		return NULL;
+	if (!__rpc_fd2sockinfo(fd, &si)) {
+		warnx("svc_vc_create: __rpc_fd2sockinfo failed");
+		goto cleanup_svc_vc_create;
+	}
 	r->sendsize = __rpc_get_t_size(si.si_af, si.si_proto, (int)sendsize);
 	r->recvsize = __rpc_get_t_size(si.si_af, si.si_proto, (int)recvsize);
 	r->maxrec = __svc_maxrec;

